### PR TITLE
Support v5

### PR DESCRIPTION
**Pull request by MicheleBertoli**
Following the discussion in #13, this PR enables the support for `react-styleguidist` v5.
So we can make the users happy, while we build a proper plugin.

### CODE REVIEW

**Comment by ForbesLindesay** [comment]
I just tried this out and I get the error:

> Module build failed: Error: Couldn't find preset "es2015" relative to directory "cwd/node_modules/snapguidist/src/components"

I'm using a different preset, I think it should just use the preset from .babelrc

File: src/core/updateWebpackConfig.js [hunk 1]
```diff
@@ -13,31 +13,25 @@ const updateWebpackConfig = (webpackConfig, env, serverInfo) => {
...
       test: /\.jsx?$/,
       include: srcFolder,
-      loader: 'babel',
+      loader: 'babel-loader',
       query: {
+        babelrc: false,
         presets: [
           'es2015',
           'react',
         ],
```
Comment (ForbesLindesay) [inline src/core/updateWebpackConfig.js]: I think this is where the problem is.  I don't think the babel loader should be specified in snapguidist, we should use the user's .babelrc file.

Comment (ForbesLindesay) [inline src/core/updateWebpackConfig.js]: Probably best to transpile the source code of snapguidist first.

**Comment by ForbesLindesay** [comment]
I was able to make this work by adding the following to my snapguidist config:

```js
  dangerouslyUpdateWebpackConfig(webpackConfig) {
    webpackConfig.module.loaders.forEach(loader => {
      if (loader.loader === 'babel-loader') {
        loader.query = undefined;
      }
    });
    return webpackConfig;
  },
```

**Comment by MicheleBertoli** [comment]
Thank you very much for looking into this @ForbesLindesay 
I added the `babel-loader` as a dependency, but I forgot to add the presets (they were present in the previous version of Styleguidist).

I guess it's OK for Snapguidist to use its own loader (for its own files) for now.
In the near future, we'll build it as a plugin - this is more a quick win to make it work with v5.

**Comment by MicheleBertoli** [comment]
Thank you very much for looking into this @ForbesLindesay 
I added the `babel-loader` as a dependency, but I forgot to add the presets (they were present in the previous version of Styleguidist).

I guess it's OK for Snapguidist to use its own loader+babelrc (for its own files) for now.
In the near future, we'll build it as a plugin - this is more a quick win to make it work with v5.

**Comment by MicheleBertoli** [comment]
Thank you very much for looking into this @ForbesLindesay 
I added the `babel-loader` as a dependency, but I forgot to add the presets (they were present in the previous version of Styleguidist).

I guess it's OK for Snapguidist to use its own loader+presets (for its own files) for now.
In the near future, we'll build it as a plugin - this is more a quick win to make it work with v5.

File: src/core/updateWebpackConfig.js (hunk 1 above)
Comment (sapegin) [inline src/core/updateWebpackConfig.js]: Transpiling is actually the only possible option. Relying on loaders *will* cause issues.

Comment (MicheleBertoli) [inline src/core/updateWebpackConfig.js]: Thanks @sapegin, but I'm not sure I get why:
> Relying on loaders will cause issues.

Can you please briefly explain me what's going to be the problem?

Comment (sapegin) [inline src/core/updateWebpackConfig.js]: I think it’s mostly explained here: https://react-styleguidist.js.org/docs/development.html

Eventually you’ll have a conflict with user’s loaders: you either break their code or they break yours.

Comment (MicheleBertoli) [inline src/core/updateWebpackConfig.js]: Got it, thanks @sapegin. I'll transpile our components.
And thanks again @ForbesLindesay for pointing this out.

**Comment by ForbesLindesay** [comment]
In the short term, you could add node-modules/snapguidist to the regex for the snapguidist babel loader, and add the presets as dependencies. Then a second babel loader that just reads babel etc can be added for user files.

**Comment by MicheleBertoli** [comment]
Thanks @ForbesLindesay, the loader should be already looking into [the source folder of Snapguidist only](https://github.com/styleguidist/snapguidist/blob/8b75cae50f6063f3bce6d3d8f32ec73ea7c3f5ca/src/core/updateWebpackConfig.js#L15), and I pushed a new version with [the presets as dependencies](https://github.com/styleguidist/snapguidist/pull/15/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R46) this morning.
I would expect it to work for your use case now.

**Comment by MicheleBertoli** [comment]
I pushed a new version which adds a build step to transpile the components.
It doesn't use any loader, except style+css (which I guess is OK for now), and it should work with any configuration.
Thanks @ForbesLindesay @sapegin.

**Comment by MicheleBertoli** [comment]
I updated a few more deps, improved a couple of functionalities and tried the package with different projects and configuration.
It seems to work fine.

**Pull request merged**